build/makelib/common.mk

```diff
@@ -74,7 +74,7 @@ endif
 # set the version number. you should not need to do this
 # for the majority of scenarios.
 ifeq ($(origin VERSION), undefined)
-VERSION := $(shell git describe --dirty --always --tags)
+VERSION := $(shell git describe --dirty --always --tags | sed 's/-g/./g')
```

Comment:
does this not have any cascading effect on any of our other scripts?

what would a new version number that isn't an official build look like? instead of something like `v0.5.0-12-g2d4fa06` what would it look like?